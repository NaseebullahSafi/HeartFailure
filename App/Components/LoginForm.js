import Ionicons from 'react-native-vector-icons/Feather';
import firebase from 'react-native-firebase';
import React from 'react'
import { StyleSheet, Text, TextInput, TouchableOpacity, TouchableWithoutFeedback, View, Keyboard } from 'react-native'
import {Field, reduxForm} from 'redux-form';
import CheckBox from 'react-native-check-box';


const onLogin = creds => {
  const {email, password} = creds;
  //console.log("props123 ", values, nav);
  firebase.auth().signInWithEmailAndPassword(email, password).then((user) => {
    /*
    * User is logged in
    *
    * */
    // if (user) navigation.navigate('drawerStack');
  }).catch(err => {
    const { code, message } = err;
    console.log("not loggedin, ", message);
  });
};

const renderInput = ({placeholder, input: { onChange, ...restInput }}) => {
  return <TextInput placeholder={placeholder} placeholderTextColor={"#bccad0"} style={styles.input} onChangeText={onChange} {...restInput} underlineColorAndroid="transparent" />
};

const LoginForm = props => {
  //console.log(props);
  const { handleSubmit } = props;

  return (
    <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
      <View style={styles.container}>
        <View style={styles.centerContainer}>

          <View style={styles.topContainer}>
            <Ionicons name="activity" size={63} color="#bccad0" />
            <Text style={styles.topContainerTxt}>Sign into your account.</Text>
          </View>

          <View style={styles.middleContainer}>
            <Field style={styles.input} name="email" placeholder="Email" component={renderInput} />
            <View style={{position: 'relative'}}>
              <Ionicons name="eye" size={18} color="rgba(188,202,208, 0.5)" style={{position: 'absolute', right: 15, top: 21}} />
              <Field style={[styles.input, {marginBottom: 40}]} name="password" placeholder="Password" component={renderInput} />
              <View style={{flexDirection: 'row', alignItems: 'center', justifyContent: 'space-between', marginTop: 10}}>
                <View style={{flexDirection: 'row', alignItems: 'center'}}>
                  <CheckBox
                    style={{marginRight: 4}}
                    checkBoxColor={'#bccad0'}
                    isChecked={false}
                    onClick={() => true}
                  />
                  <Text style={{color: '#bccad0'}}>Remember me</Text>
                </View>
                <View>
                  <Text style={{color: '#bccad0'}}>Forgot Pass?</Text>
                </View>
              </View>
            </View>
          </View>

          <View style={styles.bottomContainer}>
            <TouchableOpacity style={styles.loginBtn} onPress={handleSubmit(onLogin)}>
              <Text style={{textAlign: 'center', fontSize: 18, fontWeight: 'bold', color: 'white', letterSpacing: 2}}>L O G I N</Text>
            </TouchableOpacity>

            <View style={{flexDirection: 'row', alignItems: 'center'}}>
              <Text style={{textAlign: 'center', fontSize: 14, fontWeight: 'bold', color: '#bccad0', marginRight: 10}}>Don't have an account?</Text>
              <TouchableOpacity style={styles.signUp} onPress={() => props.navigation.navigate('SignUp')}>
                <Text style={{fontSize: 14, color: '#E67D8F', fontWeight: 'bold'}}>Sign Up</Text>
              </TouchableOpacity>
            </View>

          </View>

        </View>
      </View>
    </TouchableWithoutFeedback>
  )
}

export default reduxForm({
  form: 'test'
})(LoginForm)

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center', // Used to set Text Component Vertically Center
    alignItems: 'center', // Used to set Text Component Horizontally Center
  },
  centerContainer: {
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center'
  },
  topContainer: {
    alignItems: 'center',
    justifyContent: 'center'
  },
  topContainerTxt: {
    marginTop: 10,
    textAlign: 'center',
    fontSize: 17,
    color: '#bccad0'
  },
  middleContainer: {
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 10,
    marginBottom:10
  },
  input: {
    fontSize: 15,
    color: "#aab8be",
    height: 40,
    width: 280,
    paddingLeft: 15,
    backgroundColor: 'rgba(188,202,208, 0.15)',
    marginTop: 10,
    borderRadius: 5
  },
  bottomContainer: {
    flexDirection: 'column',
    marginTop: 30,
    alignItems: 'center'
  },
  loginBtn: {
    backgroundColor: '#E67D8F',
    alignSelf: 'stretch',
    borderRadius: 5,
    padding: 12,
    width: 280,
    height: 50,
    elevation: 2,
    marginBottom: 20
  },
  signUp: {

  }
});








/*
TODO => Uncomment below for backup
 */

// import Ionicons from 'react-native-vector-icons/Feather';
// import firebase from 'react-native-firebase';
// import React from 'react'
// import {
//   StyleSheet,
//   Text,
//   TextInput,
//   TouchableOpacity,
//   View
// } from 'react-native'
//
// import {Field, reduxForm} from 'redux-form';
//
//
//

// const onLogin = creds => {
//   const {email, password} = creds;
//   //console.log("props123 ", values, nav);
//   firebase.auth().signInWithEmailAndPassword(email, password).then((user) => {
//     /*
//     * User is logged in
//     *
//     * */
//     // if (user) navigation.navigate('drawerStack');
//   }).catch(err => {
//     const { code, message } = err;
//     console.log("not loggedin, ", message);
//   });
// };
//
// const renderInput = ({placeholder, input: { onChange, ...restInput }}) => {
//   return <TextInput placeholder={placeholder} style={styles.input} onChangeText={onChange} {...restInput} />
// };

// const LoginForm = props => {
//   const { handleSubmit } = props;
//
//   return (
//     <View style={styles.container}>
//       <View>
//         <View style={styles.elementAlignLeft}>
//           <Ionicons name="activity" size={33} color="#8F9CAE" />
//         </View>
//       </View>
//
//       <View>
//         <View style={styles.elementAlignLeft}>
//           <Text style={[styles.elementAlignLeft, styles.welcomeFontOne, {fontSize: 35}]}>
//             Welcome Back!{'\n'}
//             <Text style={[styles.elementAlignLeft, styles.welcomeFontOne, {color: '#8F9CAE'}]}>
//               Login to continue
//             </Text>
//           </Text>
//         </View>
//       </View>
//
//       <View>
//         <Field style={styles.input} name="email" placeholder="Email" component={renderInput} />
//         <Field style={[styles.input, {marginBottom: 40}]} name="password" placeholder="Password" component={renderInput} />
//       </View>
//
//       <View>
//         <View style={styles.loginBtnContainer}>
//           <TouchableOpacity style={styles.button} onPress={handleSubmit(onLogin)}>
//             <Text style={{fontSize: 30, color: '#17a2e0'}}>Login</Text>
//           </TouchableOpacity>
//         </View>
//       </View>
//
//       <View>
//         <View style={styles.elementAlignLeft}>
//           <Text style={[styles.elementAlignLeft, styles.welcomeFontOne, {marginRight: 4}]}>
//             Don't have an account? <Text style={[styles.elementAlignLeft, styles.welcomeFontOne, {color: '#8F9CAE'}]}> Signup</Text>
//           </Text>
//         </View>
//       </View>
//
//     </View>
//   )
// }

// export default reduxForm({
//   form: 'test'
// })(LoginForm)
//
// const styles = StyleSheet.create({
//   container: {
//     flex: 1,
//     justifyContent: 'space-between', // Used to set Text Component Vertically Center
//     alignItems: 'center', // Used to set Text Component Horizontally Center
//     paddingTop: 80,
//     paddingBottom: 80
//   },
//   elementAlignLeft: {
//     width: 280,
//     flexDirection: 'row',
//     justifyContent: 'flex-start',
//     alignItems: 'flex-start',
//     padding:0,
//   },
//   welcomeFontOne: {
//     flex: 1,
//     color: '#47535d'
//   },
//   loginBtnContainer: {
//     width: 280,
//     height: 60,
//     marginTop: 30,
//     justifyContent: 'center',
//     alignItems: 'flex-start'
//   },
//   button: {
//     alignItems: 'flex-start'
//   },
//   input: {
//     fontSize: 19,
//     height: 50,
//     width: 280,
//     marginTop: 10
//   }
// });
