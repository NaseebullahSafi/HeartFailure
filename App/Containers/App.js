import DebugConfig from '../Config/DebugConfig'
import React, { Component } from 'react'
import { Provider } from 'react-redux'
import RootContainer from './RootContainer'
import createStore from '../Redux'
import {MenuProvider} from "react-native-popup-menu";

// create our store
const store = createStore();

/**
 * Provides an entry point into our application.  Both index.ios.js and index.android.js
 * call this component first.
 *
 * We create our Redux store here, put it into a provider and then bring in our
 * RootContainer.
 *
 * We separate like this to play nice with React Native's hot reloading.
 */
class App extends Component {

  constructor(props) {
    super(props);
  }

  render () {

    return (
      <MenuProvider>
        <Provider store={store}>
          <RootContainer />
        </Provider>
      </MenuProvider>
    );
  }
}

// allow reactotron overlay for fast design in dev mode
export default DebugConfig.useReactotron
  ? console.tron.overlay(App)
  : App
