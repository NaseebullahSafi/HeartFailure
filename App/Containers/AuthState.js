import React, { Component } from 'react'

import firebase from 'react-native-firebase';
import LaunchScreen from "./LaunchScreen";
import OpeningLogoLoader from "./OpeningLogoLoader";

export default class AuthState extends Component {

  constructor() {
    super();

    this.state = {
      loading: true
    }
  }

  /*
* Listen for any state changes and update the component state
* */
  componentDidMount() {
    this.authSubscription = firebase.auth().onAuthStateChanged( user => this.setState({ loading: false, user }) );
  }

  componentWillUnmount() {
    this.authSubscription();
  }

  render () {
    // If the user has not authenticated
    if (this.state.loading)
      return null;

    if (this.state.user)
      return <LaunchScreen/>;

    return <OpeningLogoLoader/>;

  }

}
