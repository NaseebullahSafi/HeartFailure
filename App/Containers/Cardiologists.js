import React, {Component} from 'react';
import {Text, View, TouchableOpacity, StyleSheet, TouchableHighlight, Image, Dimensions, ScrollView} from "react-native";
import Ionicons from "react-native-vector-icons/Feather";
import Svg, { Line, G, Path } from 'react-native-svg';
import _ from 'lodash';
import firebase from 'react-native-firebase';
import Database from '../Components/Database'
import User from '../Components/User';

export default class Cardiologists extends Component {
  constructor(props) {
    super(props);
    this.state = {
      User    : null,
      doctors : null
    };

    this.usersRef = firebase.app().database().ref('/Users');
    this.fetchDoctors = this.fetchDoctors.bind(this);
  }

  componentDidMount() {
    this.fetchDoctors(this.usersRef);
  }

  fetchDoctors = (userRef) => {
    User().then(user => {
      userRef.on('value', (snap) => {
        if (snap.val()) this.setState({
          type        : snap.val()[user.uid].type,
          name        : snap.val()[user.uid].name,
          healthAlert : _.has(snap.val()[user.uid], "healthAlrt") ? snap.val()[user.uid].healthAlert : "",
          doctors : JSON.parse(JSON.stringify(snap.val()), (k, v) => !v.type || (
            v.type === 'Doctor' && (_.has(snap.val()[user.uid], "Doctors") ? !_.has(snap.val()[user.uid].Doctors, k) : true )
          ) ? v : void 0)
        }, () => {})
      })
    });
  };

  doctors = () => {
    return this.state.doctors ? (
      Object.keys(this.state.doctors).map((uid, i) => {
        return (
          <View style={styles.doctorContainer} key={uid}>
            <View style={styles.imgCircleContainer}>
              <View>
                <Image style={styles.userImg} source={require('../Images/Naseebullah.jpg')} resizeMode="contain"/>
                <View style={styles.imgOverlay} />
              </View>
            </View>

            <View>
              <Text style={{textAlign: 'center', fontSize: 15, color: '#bccad0'}}>{this.state.doctors[uid].name}</Text>
              <Text style={{textAlign: 'center', fontSize: 12, color: 'rgba(188, 202, 208, 0.7)'}}>{this.state.doctors[uid].profession}</Text>
            </View>

            <TouchableOpacity style={styles.folllowBtn} onPress={() => Database.followDoctor(
              this.state.doctors[uid].name, this.state.doctors[uid].profession, uid,
              this.state.type, this.state.healthAlert
            )}>
              <Text style={{color: 'white', fontSize: 13, fontWeight: 'bold', textAlign: 'center'}}>Follow</Text>
            </TouchableOpacity>

          </View>
        )
      })
    ): null;
  };

  render() {
    return (
      <View style={styles.mainContainer}>
        <View style={styles.topSection}>
          <View style={styles.navigationBar}>
            <TouchableHighlight onPress={() => this.props.navigation.navigate('DrawerOpen')} activeOpacity={1.0} underlayColor="rgba(253,138,94,0)">
              <Svg height="24" width="24">
                <Line fill="none" stroke="#dfe0e1" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" x1="3" y1="12" x2="21" y2="12"/>
                <Line fill="none" stroke="#dfe0e1" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" x1="10.208" y1="6" x2="21" y2="6"/>
                <Line fill="none" stroke="#dfe0e1" strokeWidth="2" strokeLinecap="round" strokeLinejoin="round" x1="3" y1="18" x2="13.791" y2="18"/>
              </Svg>
            </TouchableHighlight>
          </View>
        </View>
        <ScrollView showsVerticalScrollIndicator={false}>
          <View style={{flexDirection: 'row', flexWrap: 'wrap', justifyContent: 'center', alignItems: 'center', backgroundColor: '#f7f8f9'}}>

            {this.doctors()}

          </View>
        </ScrollView>
      </View>
    )
  }
}


const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    flexDirection: 'column',
    backgroundColor: '#f7f8f9',
    justifyContent: 'flex-start',
  },
  topSection: {
    flexDirection: 'column',
    backgroundColor: 'white',
    padding:10,
    position: 'relative',
    elevation: 0.2
  },
  navigationBar: {
    flexDirection: 'row',
    alignItems: 'stretch',
    justifyContent: 'space-between',
    padding: 10,
  },
  doctorContainer: {
    padding: 10,
    paddingTop: 40,
    paddingBottom: 20,
    width: 300,
    backgroundColor: 'white',
    margin: 10,
    borderRadius: 8,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center'
  },
  imgCircleContainer: {
    position: 'relative',
    borderRadius: 80,
    height: 70,
    width: 70,
    borderColor: 'white',
    backgroundColor: 'white',
    alignItems: 'center',
    overflow: 'hidden',
    elevation: 10,
    justifyContent: 'center',
    marginBottom: 10
  },
  userImg: {
    borderRadius: 300,
    height: 70,
    width: 70,
  },
  imgOverlay: {
    position: 'absolute',
    backgroundColor: 'rgba(247, 248, 249, 0.7)',
    opacity: 0.5,
    borderRadius: 300,
    height: 70,
    width: 70,
  },
  folllowBtn: {
    marginTop: 10,
    backgroundColor: '#2fcc71',
    width: 100,
    padding: 5,
    borderRadius: 5,
    elevation: 3,
  }
});
