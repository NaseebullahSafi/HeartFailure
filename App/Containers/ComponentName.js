import React, {Component} from 'react';

export default class ComponentName extends Component {
  constructor(props) {
    super(props);
    this.state = {...};
  }

  // called before the component is mounted
  componentWillMount() {};

  // called when the component is mounted
  componentDidMount = () => {};

  // called just before the component gets unmounted
  componentWillUnmount = () => {};

  // called when new props come
  componentWillReceiveProps = nextProps => {};

  // checks the props, if any changes, the component is re-rendered
  shouldComponentUpdate = ( nextProps, nextState, nextContext ) => {};

  // called just before it is updated
  componentWillUpdate = ( nextProps, nextState ) => {};

  // called when it is updated
  componentDidUpdate = (prevProps, prevState, prevContext) => {};

  // method that consists of the UI
  render() {
    return (null)
  }
};
