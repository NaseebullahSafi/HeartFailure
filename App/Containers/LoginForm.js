// import React from 'react'
//
// // predefined components of React
// import {
//   Text, TextInput, TouchableOpacity,
//   TouchableWithoutFeedback, View, Keyboard
// } from 'react-native'
//
// // vector icons package
// import Ionicons from 'react-native-vector-icons/Feather';
//
// // firebase database
// import firebase from 'react-native-firebase';
//
// // form fields from redux
// import {Field, reduxForm} from 'redux-form';
//
// // checkbox component 3rd party
// import CheckBox from 'react-native-check-box';
//
// // styles for this component
// import styles from '../Containers/Styles/LoginFormStyles'
//
// /**
//  * This will login the user with their email and password
//  * ==============================================================
//  * @param credentials
//  */
// const onLogin = credentials => {
//   // to login, we need the email and password of the user
//   const {email, password} = credentials;
//
//   // we call the authentication mechanism from firebase and pass the creds
//   firebase.auth()
//     .signInAndRetrieveDataWithEmailAndPassword(email, password)
//     .catch(err => ({ code, message } = err));
// };
//
// /**
//  * Converts the normal field to actual input field from redux
//  * ==============================================================
//  * @param secureTextEntry
//  * @param password
//  * @param placeholder
//  * @param onChange
//  * @param restInput
//  * @return {XML}
//  */
// const renderInput = (
//   {secureTextEntry, password, placeholder, input: { onChange, ...restInput }}) => {
//   return (
//     <TextInput
//       secureTextEntry       = {secureTextEntry}
//       password              = {password}
//       placeholder           = {placeholder}
//       placeholderTextColor  = {"#bccad0"}
//       style                 = {styles.input}
//       onChangeText          = {onChange}
//       underlineColorAndroid = "transparent"
//       {...restInput}
//     />
//   )
// };
//
// /**
//  * The container of the form is created here, with their inputs
//  * ==============================================================
//  * @param props
//  * @return {XML}
//  * @constructor
//  */
// export const LoginForm = props => {
//   const { handleSubmit } = props;
//
//   return (
//     <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
//       <View style={styles.container}>
//         <View style={styles.centerContainer}>
//
//           <View style={styles.topContainer}>
//             <Ionicons name="activity" size={63} color="#bccad0" />
//             <Text style={styles.topContainerTxt}>Sign into your account.</Text>
//           </View>
//
//           <View style={styles.middleContainer}>
//             <Field
//               style={styles.input}
//               name="email"
//               placeholder="Email"
//               component={renderInput}
//             />
//             <View style={styles.positionRelative}>
//               <Ionicons
//                 name="eye"
//                 size={18}
//                 color="rgba(188,202,208, 0.5)"
//                 style={styles.ioniconsStyle} />
//               <Field
//                 style={[styles.input, {marginBottom: 40}]}
//                 secureTextEntry={true}
//                 password={true}
//                 name="password"
//                 placeholder="Password"
//                 component={renderInput}
//               />
//               <View style={styles.checkBoxContainer}>
//                 <View style={styles.checkBoxContainerSub}>
//                   <CheckBox
//                     style={{marginRight: 4}}
//                     checkBoxColor={'#bccad0'}
//                     isChecked={false}
//                     onClick={() => true}
//                   />
//                   <Text style={{color: '#bccad0'}}>Remember me</Text>
//                 </View>
//                 <View>
//                   <Text style={{color: '#bccad0'}}>Forgot Pass?</Text>
//                 </View>
//               </View>
//             </View>
//           </View>
//
//           <View style={styles.bottomContainer}>
//             <TouchableOpacity style={styles.loginBtn} onPress={handleSubmit(onLogin)}>
//               <Text style={styles.logoutTxt}>L O G I N</Text>
//             </TouchableOpacity>
//
//             <View style={styles.dontHaveAnAccountContainer}>
//               <Text style={styles.dontHaveAccountTxt}>Don't have an account?</Text>
//               <TouchableOpacity onPress={() => props.navigation.navigate('SignUp')}>
//                 <Text style={styles.signUpTxt}>Sign Up</Text>
//               </TouchableOpacity>
//             </View>
//
//           </View>
//
//         </View>
//       </View>
//     </TouchableWithoutFeedback>
//   )
// };
//
// export default reduxForm({form: 'test'})(LoginForm)
import React, {Component} from 'react'

// predefined components of React
import {
  Text, TextInput, TouchableOpacity,
  TouchableWithoutFeedback, View, Keyboard
} from 'react-native'

// vector icons package
import Ionicons from 'react-native-vector-icons/Feather';

// firebase database
import firebase from 'react-native-firebase';

// form fields from redux
import {Field, reduxForm} from 'redux-form';

// checkbox component 3rd party
import CheckBox from 'react-native-check-box';

// styles for this component
import styles from './Styles/LoginFormStyles'

export class LoginForm extends Component {
  constructor(props) {
    super(props);
  }

  /**
   * This will login the user with their email and password
   * ==============================================================
   * @param credentials
   */
  onLogin = credentials => {
    // to login, we need the email and password of the user
    const {email, password} = credentials;

    // we call the authentication mechanism from firebase and pass the creds
    firebase.auth()
      .signInAndRetrieveDataWithEmailAndPassword(email, password)
      .catch(err => ({ code, message } = err));
  };

  /**
   * Converts the normal field to actual input field from redux
   * ==============================================================
   * @param secureTextEntry
   * @param password
   * @param placeholder
   * @param onChange
   * @param restInput
   * @return {XML}
   */
  renderInput = ({secureTextEntry, password, placeholder, input: { onChange, ...restInput }}) => {
    return (
      <TextInput
        secureTextEntry       = {secureTextEntry}
        password              = {password}
        placeholder           = {placeholder}
        placeholderTextColor  = {"#bccad0"}
        style                 = {styles.input}
        onChangeText          = {onChange}
        underlineColorAndroid = "transparent"
        {...restInput}
      />
    )
  };

  /**
   * The container of the form is created here, with their inputs
   * ==============================================================
   * @return {XML}
   * @constructor
   */
  render() {
    const { handleSubmit, navigation } = this.props;

    return (
      <TouchableWithoutFeedback onPress={Keyboard.dismiss} accessible={false}>
        <View style={styles.container}>
          <View style={styles.centerContainer}>

            <View style={styles.topContainer}>
              <Ionicons name="activity" size={63} color="#bccad0" />
              <Text style={styles.topContainerTxt}>Sign into your account.</Text>
            </View>

            <View style={styles.middleContainer}>
              <Field
                style={styles.input}
                name="email"
                placeholder="Email"
                component={this.renderInput}
              />
              <View style={styles.positionRelative}>
                <Ionicons
                  name="eye"
                  size={18}
                  color="rgba(188,202,208, 0.5)"
                  style={styles.ioniconsStyle} />
                <Field
                  style={[styles.input, {marginBottom: 40}]}
                  secureTextEntry={true}
                  password={true}
                  name="password"
                  placeholder="Password"
                  component={this.renderInput}
                />
                <View style={styles.checkBoxContainer}>
                  <View style={styles.checkBoxContainerSub}>
                    <CheckBox
                      style={{marginRight: 4}}
                      checkBoxColor={'#bccad0'}
                      isChecked={false}
                      onClick={() => true}
                    />
                    <Text style={{color: '#bccad0'}}>Remember me</Text>
                  </View>
                  <View>
                    <Text style={{color: '#bccad0'}}>Forgot Pass?</Text>
                  </View>
                </View>
              </View>
            </View>

            <View style={styles.bottomContainer}>
              <TouchableOpacity style={styles.loginBtn} onPress={handleSubmit(this.onLogin)}>
                <Text style={styles.logoutTxt}>L O G I N</Text>
              </TouchableOpacity>

              <View style={styles.dontHaveAnAccountContainer}>
                <Text style={styles.dontHaveAccountTxt}>Don't have an account?</Text>
                <TouchableOpacity onPress={() => navigation.navigate('SignUp')}>
                  <Text style={styles.signUpTxt}>Sign Up</Text>
                </TouchableOpacity>
              </View>

            </View>

          </View>
        </View>
      </TouchableWithoutFeedback>
    )
  }
}

export default reduxForm({form: 'loginForm'})(LoginForm)
