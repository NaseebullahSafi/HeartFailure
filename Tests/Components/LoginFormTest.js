// need to import React itself
import React from 'react'

// everything fro react native
import 'react-native'

// required for mocking modules
import Enzyme,{ shallow } from 'enzyme'

// needed to pass components to the renderer
import renderer from 'react-test-renderer'

import {Field, reduxForm} from "redux-form"

// adapter is needed for this react version
import Adapter from 'enzyme-adapter-react-16';

// the component that will be tested
import {LoginForm} from "../../App/Containers/LoginForm";

import { createStore } from 'redux';

import { Provider } from 'react-redux';

const store = createStore(() => ({}));

Enzyme.configure({ adapter: new Adapter() });

const Decorated = reduxForm({ form: 'testForm' })(LoginForm);

describe('Testing <LoginForm /> component', () => {
  // create an image of the component, and match it with the snapshot
  const __Decorated = (
    <Provider store={store}>
      <Decorated handleSubmit={jest.fn()}/>
    </Provider>
  );

  it('renders as expected', () => {
    const tree = renderer.create(__Decorated).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('Should render the input fields', () => {
    const wrapper = shallow(<LoginForm handleSubmit={jest.fn()} />);
    const inst = wrapper.instance().renderInput({input: jest.fn()});
    expect(inst)
  });

  it('Should render the component', () => {
    const wrapper = shallow(<LoginForm handleSubmit={jest.fn()} />);
    const inst = wrapper.instance().render()
    expect(inst)
  });
});

