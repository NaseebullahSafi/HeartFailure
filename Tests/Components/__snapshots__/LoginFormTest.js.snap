// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <LoginForm /> component renders as expected 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={false}
  hitSlop={undefined}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID={undefined}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#bccad0",
              "fontSize": 63,
            },
            undefined,
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#bccad0",
            "fontSize": 17,
            "marginTop": 10,
            "textAlign": "center",
          }
        }
      >
        Sign into your account.
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        name="email"
        onBlur={[Function]}
        onChangeText={[Function]}
        onDragStart={[Function]}
        onDrop={[Function]}
        onFocus={[Function]}
        password={undefined}
        placeholder="Email"
        placeholderTextColor="#bccad0"
        secureTextEntry={undefined}
        style={
          Object {
            "backgroundColor": "rgba(188,202,208, 0.15)",
            "borderRadius": 5,
            "color": "#aab8be",
            "fontSize": 15,
            "height": 40,
            "marginTop": 10,
            "paddingLeft": 15,
            "width": 280,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        style={
          Object {
            "position": "relative",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "rgba(188,202,208, 0.5)",
                "fontSize": 18,
              },
              Object {
                "position": "absolute",
                "right": 15,
                "top": 21,
              },
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
        <TextInput
          allowFontScaling={true}
          name="password"
          onBlur={[Function]}
          onChangeText={[Function]}
          onDragStart={[Function]}
          onDrop={[Function]}
          onFocus={[Function]}
          password={true}
          placeholder="Password"
          placeholderTextColor="#bccad0"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(188,202,208, 0.15)",
              "borderRadius": 5,
              "color": "#aab8be",
              "fontSize": 15,
              "height": 40,
              "marginTop": 10,
              "paddingLeft": 15,
              "width": 280,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "marginTop": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginRight": 4,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "process": [Function],
                    }
                  }
                  style={
                    Object {
                      "tintColor": "#bccad0",
                    }
                  }
                />
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#bccad0",
                }
              }
            >
              Remember me
            </Text>
          </View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#bccad0",
                }
              }
            >
              Forgot Pass?
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "marginTop": 30,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#E67D8F",
            "borderRadius": 5,
            "elevation": 2,
            "height": 50,
            "marginBottom": 20,
            "opacity": 1,
            "padding": 12,
            "width": 280,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
              "letterSpacing": 2,
              "textAlign": "center",
            }
          }
        >
          L O G I N
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#bccad0",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginRight": 10,
              "textAlign": "center",
            }
          }
        >
          Don't have an account?
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#E67D8F",
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            Sign Up
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
